src/m365/entra/commands/group/group-user-remove.spec.ts

```diff
@@ -0,0 +1,346 @@
+  });
+
+  it('fails validation if groupId is not a valid GUID', async () => {
+    const actual = await command.validate({ options: { groupId: 'foo', ids: userIds[0] } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if ids contains an invalid GUID', async () => {
+    const actual = await command.validate({ options: { groupId: groupId, ids: `${userIds[0]},foo` } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if userNames contains an invalid UPN', async () => {
+    const actual = await command.validate({ options: { groupId: groupId, userNames: `${userUpns[0]},foo` } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if role is not a valid role', async () => {
+    const actual = await command.validate({ options: { groupId: groupId, ids: userIds.join(','), role: 'foo' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation when all required parameters are valid with ids', async () => {
+    const actual = await command.validate({ options: { groupId: groupId, ids: userIds.join(',') } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when all required parameters are valid with ids with leading spaces', async () => {
+    const actual = await command.validate({ options: { groupId: groupId, ids: userIds.map(i => ' ' + i).join(','), role: 'Member' } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when all required parameters are valid with names', async () => {
+    const actual = await command.validate({ options: { groupDisplayName: 'IT department', userNames: userUpns.join(',') } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when all required parameters are valid with names with trailing spaces', async () => {
+    const actual = await command.validate({ options: { groupDisplayName: 'IT department', userNames: userUpns.map(u => u + ' ').join(','), role: 'Owner' } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('prompts before removing the specified users when confirm option not passed', async () => {
+    const confirmationStub = sinon.stub(cli, 'promptForConfirmation').resolves(false);
+
+    await command.action(logger, { options: { groupDisplayName: 'IT department', ids: userIds.join(',') } });
+
+    assert(confirmationStub.calledOnce);
+  });
+
+  it('aborts removing users when prompt not confirmed', async () => {
+    sinon.stub(cli, 'promptForConfirmation').resolves(false);
+
+    const postSpy = sinon.stub(request, 'post').resolves();
+
+    await command.action(logger, { options: { groupId: groupId, ids: userIds.join(',') } });
+    assert(postSpy.notCalled);
+  });
+
+  it('successfully removes owners and members from the group with ids after confirming prompt', async () => {
```

Comment:
Let's ensure that in every test we've got an assert. Otherwise, it's possible that a test will pass while the code is not working as intended.

Reply:
There is an assert on line 147 right?